plugins/auth-backend/src/providers/google/provider.ts

```diff
@@ -38,30 +37,37 @@ import {
   makeProfileInfo,
   PassportDoneCallback,
 } from '../../lib/passport';
-import { AuthProviderFactory, RedirectInfo } from '../types';
+import {
+  AuthProviderFactory,
+  RedirectInfo,
+  SignInResolver,
+  ProfileTransform,
+} from '../types';
 import { TokenIssuer } from '../../identity';
 
 type PrivateInfo = {
   refreshToken: string;
 };
 
 type Options = OAuthProviderOptions & {
-  logger: Logger;
-  identityClient: CatalogIdentityClient;
+  signInResolver?: SignInResolver<OAuthResult>;
```

Comment:
I do not understand the flow for this. Why are both `signInResolver` and a `profileTransform` required ? 
The pipeline seems to be simple `OAuthResponse` (authorization_code), -> `OAuth` Result (id + acc + ref token) -> `profileTransform` (`backstageIdentity` - `token`, `id`, `ent`). Is there anything I am missing ?

Reply:
`profileTransform` is needed for all providers, it's the logic that creates the simplified profile used to show your logged in state in the frontend, e.g.

![image](https://user-images.githubusercontent.com/4984472/110470081-b605ee80-80da-11eb-90e5-5029c35a7d27.png)

The `signInResolver` is only needed for providers that one wants to use for sign-in, and it contains the logic for how to map the auth provider result to a Backstage identity.

Reply:
Could it make sense to name it `identityTransform` to be consistent with the other one?

Reply:
@Rugvip ^ thoughts on replacing `signInResolver` with `identityTransform` ?